### PR TITLE
Added invenio oauth2 server

### PR DESCRIPTION
**Pull request by monaawi**
INSPIR-1992
Added invenio oauth2 server
Added authentication to workflow author submission endpoint
Added fixture to create authentication token for local development

### CODE REVIEW

File: tests/integration/workflows/test_views.py [hunk 1]
```diff
@@ -145,38 +146,40 @@ def test_new_author_submit_with_required_fields(api_client):
...
+                "Authors"
+            ],
+            "name": {
+                "value": "Martinez, Diegpo"
+            }
         }
-    }
 
-    assert expected == obj.data
+        assert expected == obj.data
```
Comment (drjova) [inline tests/integration/workflows/test_views.py]: is this a tab?

File: tests/integration/search/test_search_facets.py [hunk 2]
```diff
@@ -74,19 +75,19 @@ def load_json_resource(resource):
...
-        api_client.get('literature/facets?q=%s' % query).data)
+        api_client.get('literature/facets?%s' % urllib.urlencode(query)).data)
 
-    query = '"a Müller"'
+    query = {'q': 'a Müller'}
     expected = load_json_resource('test_hep_facets_by_author_name_expected_2')
 
     assert expected == json.loads(
-        api_client.get('literature/facets?q=%s' % query).data)
+        api_client.get('literature/facets?%s' % urllib.urlencode(query)).data)
```
Comment (drjova) [inline tests/integration/search/test_search_facets.py]:
```suggestion
        query = '"a Müller"'
        api_client.get('literature/facets?q=%s' % urllib.urlencode(query)).data)
```?

File: tests/integration/workflows/test_views.py [hunk 3]
```diff
@@ -145,38 +146,40 @@ def test_new_author_submit_with_required_fields(api_client):
             "status": "active"
         }
     }
-    response = api_client.post('/workflows/authors', data=json.dumps(data), content_type='application/json')
-    assert response.status_code == 200
 
-    workflow_object_id = json.loads(response.data).get('workflow_object_id')
-    assert workflow_object_id is not None
+    with workflow_app.test_client() as client:
+        response = client.post('/workflows/authors', data=json.dumps(data), content_type='application/json', headers={"Authorization": "Bearer " + current_app.config["AUTHENTICATION_TOKEN"]})
```
Comment (drjova) [inline tests/integration/workflows/test_views.py]: it's more readable to move headers in a separate var

**Comment by drjova** [comment]
@monaawi we should add a test with no or invalid token

File: inspirehep/modules/fixtures/users.py [hunk 4]
```diff
@@ -176,3 +177,29 @@ def init_users_and_permissions():
...
+def init_authentication_token():
+    with db.session.begin_nested():
+        c = Client(
+            name='admin',
+            user_id=1,
+            is_internal=True,
+            is_confidential=False,
+            _default_scopes=""
+        )
+        c.gen_salt()
```
Comment (drjova) [inline inspirehep/modules/fixtures/users.py]: do we need the salt?

File: tests/integration/search/test_search_facets.py (hunk 2 above)
Comment (monaawi) [reply tests/integration/search/test_search_facets.py]: urllib.urlencode take a dictionary with the query parameters as an argument, If I give it a string it throws an error

File: inspirehep/modules/fixtures/users.py (hunk 4 above)
Comment (monaawi) [reply inspirehep/modules/fixtures/users.py]: yes, it generates the client id which we use when creating the token

File: tests/integration/workflows/test_views.py [hunk 5]
```diff
@@ -196,33 +201,62 @@ def test_update_author_submit_with_required_fields(api_client):
...
+                               headers=headers)
+        assert response.status_code == 200
 
-    workflow_object_id = json.loads(response.data).get('workflow_object_id')
-    assert workflow_object_id is not None
+        workflow_object_id = json.loads(response.data).get('workflow_object_id')
+        assert workflow_object_id is not None
 
-    obj = workflow_object_class.get(workflow_object_id)
+        obj = workflow_object_class.get(workflow_object_id)
```
Comment (drjova) [inline tests/integration/workflows/test_views.py]: tab here as well :/

**Comment by drjova** [comment]
lGTM just the weird indentation

**Comment by monaawi** [comment]
I don't know why it has the weird indentation :( I ran formatting from pycharm, I will try and fix it

**Comment by drjova** [comment]
@monaawi there are some conflicts and cannot merge it

**Pull request merged**